### DISCUSSION TITLE
Your Session has been timed out after 3 minutes.

### DISCUSSION THREAD

**Discussion by Wolbaz** [dani-garcia/vaultwarden#3721] [Q&A]
My company email last week has started taking longer to receive outside emails, approximately 3 minutes. By the time my 2FA email comes in, I get a message telling me my session has timed out. Attempting to resend the email gives the same error, I can only have it re-sent by logging in from scratch again. My current work around is to use my personal email, but I have to rely on google phone call 2FA to log into my personal email, which isn't as friendly. For context I can't have my cellphone at work.

### Deployment environment
Docker container on Unraid

* vaultwarden version:1.29.0

* Install method: Docker Container

* Clients used: web vault

* Reverse proxy and version: Traefik 2.6.7

* MySQL/MariaDB or PostgreSQL version: n/a

* Other relevant details:

### Steps to reproduce
Use SMTP 2FA, wait >= 3 minutes, session times out.

### Expected behaviour
Session doesn't time out

### Actual behaviour
Session times out

**Comment by BlackDex** [comment]
This can be configured see https://github.com/dani-garcia/vaultwarden/blob/3dbfc484a54c41d1759646444b439da06445060b/.env.template#L233

**Reply by Wolbaz** [reply]
My confusion then is that I've got it set to 600 seconds which is 10 minutes, but it's timing me out after 3 minutes.

**Comment by zacknewman** [comment]
https://github.com/dani-garcia/vaultwarden/blob/main/src/config.rs#L450C9-L450C34. If you don't feel like reading that, then set `INCOMPLETE_2FA_TIME_LIMIT` to the number of minutes you want.

FYI, you must use uppercase for the `.env` variable names even if the support string that is generated in the admin panel shows them as lowercase—and no, Vaultwarden won't complain if you have invalid variable names. The reason it is in snake_case in code is due to the [Rust style guide](https://rust-lang.github.io/api-guidelines/naming.html). Also don't be fooled by the commented out lines in `.env.template`. A vast majority of them agree with the default value, but there are some that don't. Another reason it is better to read the source code at least for `config.rs`. Don't worry about not knowing Rust either as I am sure you can "guess" what is happening.

**Comment by Wolbaz** [comment]
Since my diagnostic data was removed let me also state that my INCOMPLETE_2FA_TIME_LIMIT is also set to 10 minutes.
